React-Assmt/src/js/modules/prelogin/studentApp.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import allResults from './../../../static/files/results.json';
+import Search from './components/search.js';
+import Checkbox from './components/checkbox.js';
+import '../../../static/stylesheets/style';
+import { connect } from 'react-redux';
+import { updateData } from './actions';
+import { Link } from 'react-router';
+import { checkboxEnum, ResultType } from './components/constants.js';
+
+var total,percentage,checkValue=[];
+class StudentApp extends Component {
+
+  componentDidMount () {
+    this.props.updateData(allResults.results);
+  }
+
+  _generateTable() {
+    if (this.props.results) {
+      // Added search with redux 
+      if(this.props.updatedResults.length) {
+        var percentageResults = Object.assign([],this.props.updatedResults);
+      } else { 
+        var percentageResults = Object.assign([],this.props.results);
+      }      
+      return percentageResults.map((value, key) => {
+        total = (value.marks.english + value.marks.hindi + value.marks.mathematics);
+        percentage = parseInt(total / 3, 10);
+        if (!percentageResults.percentage) {
+          percentageResults[key].percentage= percentage;
+        }
+        return (
+          <tr key = {key} className = {percentage < 35 ? "failColor" : null}>
+            <td><Link to={{pathname: "home/", query: {param1: value.firstName, param2: value.lastName, param3: value.marks.english, param4: value.marks.hindi, param5: value.marks.mathematics, param6: total, param7:percentage }}} className="field" target="_blank">{value.firstName}</Link></td>
```

Comment:
If the passing object is big then form object outside template and then pass it.